File: frontend/src/api/binLocation.ts
import { get, post, put } from '../utils/request'
import type { Paginated, PageQuery } from '../types/enums'

export interface BinView {
  id: number
  code: string
  area: string
  rack_no: string
  layer_no: number
  column_no: number
  capacity: number
  occupancy_rate: number
  storage_requirement: string
  storage_text: string
  status: string
  status_text: string
  created_at: string
}

export interface BinContent {
  product_id: number
  product_name: string
  sku: string
  batch_no: string
  quantity: number
  owner_name: string
}

export function apiListBins(params: PageQuery) {
  return get<Paginated<BinView>>('/bin-locations', params)
}

export function apiGetBin(id: number) {
  return get<BinView>(`/bin-locations/${id}`)
}

export function apiGetBinContents(id: number) {
  return get<BinContent[]>(`/bin-locations/${id}/contents`)
}

export function apiCreateBin(data: Record<string, unknown>) {
  return post<BinView>('/bin-locations', data)
}

export function apiBatchCreateBins(data: Record<string, unknown>) {
  return post<{ created: number }>('/bin-locations/batch', data)
}

export function apiUpdateBin(id: number, data: Record<string, unknown>) {
  return put<BinView>(`/bin-locations/${id}`, data)
}

export function apiRecommendBin(productId: number, quantity: number) {
  return post<BinView>('/bin-locations/recommend', { product_id: productId, quantity })
}

File: backend/internal/model/bin_location.go
package model

import "time"

// BinLocation 库位。
type BinLocation struct {
	ID                uint      `gorm:"primaryKey" json:"id"`
	Area              string    `gorm:"size:4;index;not null" json:"area"`
	RackNo            string    `gorm:"size:16;not null" json:"rack_no"`
	LayerNo           int       `gorm:"not null" json:"layer_no"`
	ColumnNo          int       `gorm:"not null" json:"column_no"`
	Capacity          float64   `gorm:"type:numeric(10,2);not null" json:"capacity"`
	OccupancyRate     float64   `gorm:"type:numeric(6,2);default:0" json:"occupancy_rate"`
	StorageRequirement string   `gorm:"size:16;not null;default:Normal" json:"storage_requirement"`
	Status            string    `gorm:"size:16;not null;default:Available" json:"status"`
	CreatedAt         time.Time `json:"created_at"`
	UpdatedAt         time.Time `json:"updated_at"`
}
